### PR TITLE
Magento_Review: avoid using deprecated escape* methods from AbstractB…

### PR DESCRIPTION
**Pull request by sergiynezbritskiy**
### Description (*)

Avoid using deprecated escape* methods from AbstractBlock

### Related Pull Requests

https://github.com/magento/magento2/pull/31610

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Review/view/frontend/templates/helper/summary_short.phtml [hunk 1]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/** @var \Magento\Review\Block\Product\ReviewRenderer $block */
+/**
+ * @var \Magento\Review\Block\Product\ReviewRenderer $block
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/Review/view/frontend/templates/helper/summary_short.phtml]:
```suggestion
/**
 * @var \Magento\Review\Block\Product\ReviewRenderer $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/Review/view/frontend/templates/product/view/list.phtml [hunk 2]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/** @var Magento\Review\Block\Product\View\ListView $block */
+/**
+ * @var Magento\Review\Block\Product\View\ListView $block
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/Review/view/frontend/templates/product/view/list.phtml]:
```suggestion
/**
 * @var Magento\Review\Block\Product\View\ListView $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Review/view/frontend/templates/helper/summary_short.phtml (hunk 1 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Review/view/frontend/templates/helper/summary_short.phtml]: fixed

File: app/code/Magento/Review/view/frontend/templates/product/view/list.phtml (hunk 2 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Review/view/frontend/templates/product/view/list.phtml]: fixed

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests EE, WebAPI Tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please resolve conflicts

File: app/code/Magento/Review/view/frontend/templates/product/view/list.phtml (hunk 2 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Review/view/frontend/templates/product/view/list.phtml]: trigger removing "Progress: needs update" label

**Comment by ihor-sviziev** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run Database Compare

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8926](https://jira.corp.magento.com/browse/ENGCOM-8926) has been created to process this Pull Request

**Pull request closed without merging**